Quantity price attriubute

mkasztelnik opened the pull request:
Example configuration:

```
{
  "id"=>"id7",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>100, "start_price"=>1024},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
}
```

Demo:

![bedzie](https://user-images.githubusercontent.com/1265430/53246233-6baaa400-36b0-11e9-8463-3acc3e683e2a.png)

Fixes #693

agpul commented:
I have one comment. How we can realize the case when we have one offer with two price option, as I mentioned in the ticket (for eg. the same offer has different price for EGI participant and different price for others).

mkasztelnik commented:
@agpul two attributes. One for "EGI Participant" second for "outside people"

mkasztelnik commented:
You are right, but this is general remark (maybe already solved by current implementation) - @bwilk how optional attributes can be modeled in current attribute implementation?

goreck888 commented on app/models/attribute.rb:
```diff
@@ -114,6 +114,8 @@ def self.from_json(json)
       attr = Attribute::Date.new
     when "range-property"
       attr = Attribute::RangeProperty.new
+    when "quantity_price"
```
That's I mean in my review. People who will update offers will get crazy if we won't be consistent.

bwilk replied:
Agreed

mkasztelnik replied:
Renamed, now configuration should looks like:

bwilk commented:
> You are right, but this is general remark (maybe already solved by current implementation) - @bwilk how optional attributes can be modeled in current attribute implementation?

There is no way to model something _optional value_ on a attribute level (as a generic feature). Consider how would optional _select_ look like? Strange istn't it? So - _optional_ value should be modeled as a part of the specific attribute type - e.g. input

roksanaer commented:
apart from the attribute type name consistency, this PR is OK for this release. I do not perceive this
>>  I have one comment. How can we realize the case when we have one offer with two price option, as I mentioned in the ticket (for eg. the same offer has different price for EGI participant and different price for others).
as a problem. In this case we have 2 separate attributes with 2 different formulas. Both with a default value for number of participants set to zero.
Hovewer we need to better name and define the attributes meant for the specific use case shat stands behind this PR (FitSM training offers).
@agpul can you define in this PR those attributes (config in presta as a source) basing on the expample Marek gave?

agpul commented:
1. First one is FitSM [https://marketplace.docker-fid.grid.cyf-kr.edu.pl/services/fitsm](url)

* FitSM Foundation Level:
 - EGI Participant = yes
  {
  "id"=>"id1",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>80, "start_price"=>1440},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
}
 - EGI Participant = No
  {
  "id"=>"id2",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>80, "start_price"=>1680},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
}
* Advanced Level in Service Planning and Delivery:
- EGI Participant = yes
  {
  "id"=>"id1",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>160, "start_price"=>2880},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
}
 - EGI Participant = No
  {
  "id"=>"id2",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>160, "start_price"=>3360},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
}  
* Advanced Level in Service Operation and Control:
- EGI Participant = yes
  {
  "id"=>"id1",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>160, "start_price"=>2880},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
}
 - EGI Participant = No
  {
  "id"=>"id2",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>160, "start_price"=>3360},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
}  

* Expert Level:
 - EGI Participant = yes
  {
  "id"=>"id1",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>240, "start_price"=>3640},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
}
 - EGI Participant = No
  {
  "id"=>"id2",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>240, "start_price"=>4240},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
} 

* Expert Bridge: (we don't have such option now but I believe we will have it)
 - EGI Participant = yes
  {
  "id"=>"id1",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>400, "start_price"=>4250},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
}
 - EGI Participant = No
  {
  "id"=>"id2",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>400, "start_price"=>5000},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
}

agpul commented:
Second offer is (we don't have it now in MP but we need to add ) ISO 27001:

1. ISO 27001 Foundation:

* EGI Participant = yes
  {
  "id"=>"id1",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>150, "start_price"=>4400},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
  }
* EGI Participant = No
  {
  "id"=>"id2",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>150, "start_price"=>5150},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
  }

2. ISO 27001 Professional:

* EGI Participant = yes
  {
  "id"=>"id1",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>250, "start_price"=>8750},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
  }
* EGI Participant = No
  {
  "id"=>"id2",
  "type"=>"quantity_price",
  "label"=>"Price per unit",
  "config"=>{"max"=>15, "currency"=>"EUR", "step_price"=>250, "start_price"=>10250},
  "value_type"=>"integer",
  "description"=>"Choose quantity"
  }

agpul commented:
I put general attributes here. For sure we (or Mal?) need to add better labels and descriptions for each one.

mkasztelnik commented:
Updated according to the review, rebased to latest master. The open question is connected with attribute mandatory/optional issue, but as far as I understand slack conversation this can be solved right now by using default `0` default value.

roksanaer commented:
>> The open question is connected with attribute mandatory/optional issue, but as far as I understand slack conversation this can be solved right now by using default `0` default value.

true
